### DISCUSSION TITLE
How to crawl only the suggested site and not external sites

### DISCUSSION THREAD

**Discussion by benzvi8888** [safesploit/Doogle#24] [Q&A]
Seems when I checked the database the crawler go into social link and start index their content - it crawl Facebook, Twitter, LinkedIn and so on - it filled my database with it.
is there any way/option/code to limit the crawling to the site itself only?

**Comment by safesploit** [comment]
Unfortunately, I do not remember designing the code like this.

The functionality is in these functions below, but when I designed this, modularity and best practice weren't in mind. Hence, I don't believe there will be an 'easy fix' unless you're willing to rewrite most of the Crawler class.

- [getDetails](https://github.com/safesploit/doogle/blob/50304b956c92a1220dea3c26d7cb74599fc4bb56/classes/Crawler.php#L88
)
- [followLinks](https://github.com/safesploit/doogle/blob/50304b956c92a1220dea3c26d7cb74599fc4bb56/classes/Crawler.php#L166)
